### PR TITLE
Refs #27236 -- Reverted AlterIndexTogether deprecation.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-27236

### CODE REVIEW

File: docs/ref/migration-operations.txt [hunk 1]
```diff
@@ -102,9 +102,14 @@ subclass).
...
-:attr:`~django.db.models.Options.index_together` option on the ``Meta``
-subclass).
+Changes the model's set of custom indexes (the ``index_together`` option on the
+``Meta`` subclass).
+
+.. warning::
+
+    ``AlterIndexTogether`` is officially supported only for pre-Django 4.2
+    migration files. For backward compatibility reasons, it's still part of the
+    public API, and there's no plan to deprecate or remove it.
```
Comment (carltongibson) [inline docs/ref/migration-operations.txt:112]: Don't shoot me 😜... 

Is it worth having an _Unapplied only_ check (perhaps just a warning) on migrations if they use ``AlterIndexTogether`` so folks who didn't read this know not to use it? 

That would trigger if I rolled-back (or rebuilt or ...) as per the previous discussion but that seems OK. 🤔

_No_ might be the right answer 🙂

Comment (felixxm) [reply docs/ref/migration-operations.txt:112]: I would say "no" :wink: Django 4.2+ will not generate migrations with `AlterIndexTogether` anymore, and who would add it by hand :shrug:

File: docs/ref/migration-operations.txt [hunk 2]
```diff
@@ -102,16 +102,14 @@ subclass).
...
-.. deprecated:: 4.2
+.. warning::
 
-    ``AlterIndexTogether`` is deprecated in favor of
-    :class:`~django.db.migrations.operations.AddIndex`,
-    :class:`~django.db.migrations.operations.RemoveIndex`, and
-    :class:`~django.db.migrations.operations.RenameIndex` operations.
+    ``AlterIndexTogether`` is officially supported only for pre-Django 4.2
+    migration files. For backward compatibility reasons, it's still part of the
+    public API, and there's no plan to deprecate or remove it.
```
Comment (collinanderson) [inline docs/ref/migration-operations.txt:112]: I'd suggest encouraging use of supported `AddIndex` / `RemoveIndex` here.

File: docs/releases/4.2.txt [hunk 3]
```diff
@@ -308,12 +307,14 @@ Running the :djadmin:`makemigrations` command will generate a migration
 containing a :class:`~django.db.migrations.operations.RenameIndex` operation
 which will rename the existing index.
 
+The ``AlterIndexTogether`` migration operation is now officially supported only
+for pre-Django 4.2 migration files. For backward compatibility reasons, it's
+still part of the public API, and there's no plan to deprecate or remove it.
```
Comment (collinanderson) [inline docs/releases/4.2.txt:312]: I'd suggest encouraging use of supported `AddIndex` / `RemoveIndex` here.

**Comment by collinanderson** [comment]
I'm definitely in favor of this direction. Thanks!

**Comment by felixxm** [comment]
@David-Wobrock Does it work for you?

**Comment by David-Wobrock** [comment]
If I understand correctly, we deprecate `Meta.index_together`, but we leave `AlterIndexTogether` as it is for historical migrations.
Sounds good to me 👍

**Comment by felixxm** [comment]
@carltongibson Thanks for checking :+1:

**Pull request closed without merging**